pkg/backup/handlers/arango/backup/state_ready.go

```diff
@@ -56,7 +56,7 @@ func stateReadyHandler(h *handler, backup *backupApi.ArangoBackup) (*backupApi.A
 		)
 	}
 
-	if backupMeta.Available == false {
+	if !backupMeta.Available || backupMeta.NumberOfDBServers != backupMeta.NumberOfPiecesPresent {
```

Comment:
Although I studied math, I dislike expressing `not ( _ and _)` as `not _ or not _` :D maybe create a local variable `bool available = backupMeta.Available && backupMeta.NumberOfDBServers == backupMeta.NumberOfPiecesPresent`
add a comment for temporary workaround and use `!available` for the condition. Then the future you and me know what was going on there

Reply:
Check new logic